examples/ota-provider-app/ota-provider-common/DefaultUserConsentProvider.h

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <ota-provider-common/UserConsentDelegate.h>
+
+namespace chip {
+namespace ota {
+
+class DefaultUserConsentProvider : public UserConsentDelegate
+{
+public:
+    DefaultUserConsentProvider();
+
+    ~DefaultUserConsentProvider() = default;
+
+    // This method returns kGranted unless explicitly denied by the user by calling RevokeUserConsent()
+    UserConsentState GetUserConsentState(const UserConsentSubject & subject) override;
+
+    // Grant the user consent for the given subject for OTA updates
+    CHIP_ERROR GrantUserConsent(const UserConsentSubject & subject);
+
+    // Revoke the user consent for the given subject for OTA updates
+    CHIP_ERROR RevokeUserConsent(const UserConsentSubject & subject);
+
+    // Set the user consent as deferred for the given subject for OTA updates
+    CHIP_ERROR DeferUserConsent(const UserConsentSubject & subject);
```

Comment:
@tcarmelveilleux I have added `SetGlobalUserConsentState()` API which can test all the paths, should we keep these APIs for tracking individual subject or remove it?